### PR TITLE
Userstory2

### PR DESCRIPTION
**Pull request by kisa5**

### CODE REVIEW

File: components/ProjectCard.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
-import titles from "lib/db.json";
-import { v4 as uuidv4 } from "uuid";
+import { projects } from "@/lib/db.js";
 import styled from "styled-components";
+import Link from "next/link";
+import { useRouter } from "next/router";
 
 export default function ProjectCard() {
   return (
     <>
```
Comment (F-Kirchhoff) [inline components/ProjectCard.js]: You don't need these fragments. Please use them only if you want to return two or more elements "side by side" and not wrapped in a container element.

File: lib/db.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    title: "project 2",
+    tasks: [{ task: "task 1" }, { task: "task 2" }, { task: "task 3" }],
+  },
+  {
+    slug: "3",
+    id: 2,
+    title: "project 3",
+    tasks: [{ task: "task 1" }, { task: "task 2" }, { task: "task 3" }],
+  },
+];
```
Comment (F-Kirchhoff) [inline lib/db.js:20]: Nice data structure!

File: pages/project/[slug].js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { projects } from "@/lib/db.js";
+import styled from "styled-components";
+
+export default function Tasks() {
+  const router = useRouter();
+  const { slug } = router.query;
+
+  const projectIndex = projects.findIndex((project) => project.slug === slug);
+
+  const project = projects[projectIndex];
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js]: you can combine those two lines:
```
const project = projects.find(project => project.slug === slug)
```

File: pages/project/[slug].js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  if (!project) {
+    return null;
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js]: In the future you can redirect here with the router to your 404 page.

File: pages/project/[slug].js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  const { title, tasks } = project;
+
+  return (
+    <>
+      <button>
+        <Link href="/">⬅️</Link>
+      </button>
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js]: Please don't use a button for a link component.

Comment (F-Kirchhoff) [reply pages/project/[slug].js]: You can style the Link though that it looks like a button.

File: pages/project/[slug].js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      <h1>{title}</h1>
+      <ul>
+        {tasks.map(({ task }) => (
+          <li key={task}>{task}</li>
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js]: The task might not be unique. You can add an id to your tasks though since they are already objects!

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Great work! have a look at the comments. Next round it should be ready to go.

**Comment by kisa5** [comment]
Thank you Felix for your Advice! Did some changes.. 🙂

File: pages/project/[slug].js [hunk 7]
```diff
@@ -7,27 +7,54 @@ export default function Tasks() {
...
   const { slug } = router.query;
 
-  const projectIndex = projects.findIndex((project) => project.slug === slug);
-
-  const project = projects[projectIndex];
+  const project = projects.find((project) => project.slug === slug);
 
   if (!project) {
-    return null;
+    return <p>404 page</p>;
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js:13]: okay, that is good for now, consider creating a dedicated page at some later point in time

File: pages/project/[slug].js [hunk 8]
```diff
@@ -7,27 +7,54 @@ export default function Tasks() {
...
   );
 }
+
+const BackButton = styled.div`
+  margin: 5px;
+`;
+
+const Title = styled.h1`
+  top: 40px;
+  left: 15px;
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js:39]: top and left are ignored in static positioning. So you can remove it.

File: pages/project/[slug].js [hunk 9]
```diff
@@ -7,27 +7,54 @@ export default function Tasks() {
...
+  font-style: bold;
+  font-size: 17px;
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js:41]: You may want to use rem here for accessibility reasons.

File: pages/project/[slug].js [hunk 10]
```diff
@@ -7,27 +7,54 @@ export default function Tasks() {
...
+`;
+
+const TaskItem = styled.li`
+  background-color: lightgrey;
+  padding: 10px;
+  margin: 5px;
+  list-style-type: none;
+  width: 85%;
+  border-radius: 5px;
+  font-size: 15;
```
Comment (F-Kirchhoff) [inline pages/project/[slug].js:59]: you are missing the units here

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Looks much better now. Please account the css issues, then your are ready to go!

**Pull request merged**